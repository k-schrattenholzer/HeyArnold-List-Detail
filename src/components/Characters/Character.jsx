export default function Character({ character }) {
  return (
    <section>
      <h1>{character.name}</h1>
      <img src={character.image} alt={character.name} />
      <p>id:{character._id}</p>
    </section>
  )
}