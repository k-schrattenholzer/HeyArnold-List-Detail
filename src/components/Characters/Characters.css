
li {
  list-style-type: none;
}

img {
  width: 40px;
  height: 50px;
}

.characterListEl {
  grid-column: span 2;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.imageContainer {
  height: 200px;
  margin: 10px;
  padding: 20px;
  /* gap: 10em; */
}
.imageEl {
  border-radius: 85px;
  opacity: 1;
  display: block;
  width: 100%;
  height: 100%;
  transition: .5s ease;
  backface-visibility: hidden;

}

.imageContainer:hover .imageEl {
  opacity: 0.3;
}
