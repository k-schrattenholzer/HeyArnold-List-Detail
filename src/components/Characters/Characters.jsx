import { Link } from "react-router-dom";
import styles from './Characters.css'

export default function Characters({ characterList }) {
  return(

    <div className={styles.characterListEl}>
    {characterList.map((char) => 
      <li className={styles.imageContainer} key={char._id}>
        <Link to={`/character-detail/${char._id}`}>
          <img src={char.image} alt={char.name} className={styles.imageEl} />
        </Link>
        
      </li>
    )
    }
    </div>
  )
}
