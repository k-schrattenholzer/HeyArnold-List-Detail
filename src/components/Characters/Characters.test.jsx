import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Characters from './Characters.jsx';

it('should render the characters component', () => {
  const testCharacters = [
    {
      _id: 1,
      name: 'Grandma Gertie',
      image: 'https://website.com/image/gertie.jpg',
    },
    {
      _id: 2,
      name: 'Arnie',
      image: 'https://website.com/image/arnie.jpg',
    },
    {
      _id: 3,
      name: 'Stella Shortman',
      image: 'https://website.com/image/Stella-Shortman.jpg',
    },
  ]
  const { container } = render (
    <MemoryRouter>
      <Characters characterList={testCharacters}/>
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});