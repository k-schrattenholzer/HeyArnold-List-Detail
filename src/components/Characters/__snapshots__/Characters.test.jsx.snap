// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the characters component 1`] = `
<div>
  <div
    class="characterListEl"
  >
    <li
      class="imageContainer"
    >
      <a
        href="/character-detail/1"
      >
        <img
          alt="Grandma Gertie"
          class="imageEl"
          src="https://website.com/image/gertie.jpg"
        />
      </a>
    </li>
    <li
      class="imageContainer"
    >
      <a
        href="/character-detail/2"
      >
        <img
          alt="Arnie"
          class="imageEl"
          src="https://website.com/image/arnie.jpg"
        />
      </a>
    </li>
    <li
      class="imageContainer"
    >
      <a
        href="/character-detail/3"
      >
        <img
          alt="Stella Shortman"
          class="imageEl"
          src="https://website.com/image/Stella-Shortman.jpg"
        />
      </a>
    </li>
  </div>
</div>
`;
